modules/source-bookshelf/lib/destructure_request_data.js

```diff
@@ -0,0 +1,52 @@
+const sanitizeRequestData = require('./sanitize_request_data');
+
+module.exports = function(model, params) {
+  var destructured = {
+    data: params,
+    toManyRels: []
+  };
+
+  if (!params) {
+    return destructured;
+  }
+
+  var relations = params.links;
+
+  if (relations) {
+
+    // probably could be a reduce...
+    destructured = Object.keys(relations).reduce(function(result, key) {
+      if (!model.related(key)) {
```

Comment:
Can we get a TODO note here about pulling all of these things out and putting them into a library like [boom](https://github.com/hapijs/boom). I don't care about this now, since we need to get this usable, but eventually we gotta wrangle all of these into some kind of order.